app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -113,14 +115,35 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     });
 
+    const mergedScopes = mergeScopes(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
+    );
+
+    // clear per scope middleware
+    hooks.multichainMiddlewareManager.removeAllMiddleware();
+    hooks.multichainSubscriptionManager.unsubscribeAll();
+
+    // if the eth_subscription notification is in the scope and eth_subscribe is in the methods
+    // then get the subscriptionManager going for that scope
+    Object.entries(mergedScopes).forEach(([scope, scopeObject]) => {
```

Comment:
Thoughts on doing this when establishing the RPC pipeline for the connection as well?

Reply:
I think this logic should be part of a PermissionController:stateChange listener instead to handle the case where permissions are added outside of a `provider_authorize` call

Reply:
done in [a5f73ec](https://github.com/MetaMask/metamask-extension/pull/25869/commits/a5f73ec14b14baff8c49e2f9a5c8ff95e0d7f20d)